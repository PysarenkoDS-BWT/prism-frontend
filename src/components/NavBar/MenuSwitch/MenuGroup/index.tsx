import React, { useState } from 'react';
import {
  createStyles,
  MenuItem,
  Select,
  WithStyles,
  withStyles,
} from '@material-ui/core';
import {
  LayerKey,
  LayerType,
  LayerMenuGroupItem,
} from '../../../../config/types';
import { LayerDefinitions } from '../../../../config/utils';

function MenuGroup({ classes, menuGroup, toggleLayerValue }: MenuGroupProps) {
  const [activeLayer, setActiveLayer] = useState(menuGroup[0].id);

  const handleChange = (
    event: React.ChangeEvent<{ value: string | unknown }>,
  ) => {
    const layerId = event.target.value;
    const layer = LayerDefinitions[layerId as LayerKey];
    toggleLayerValue(layer, true);
    setActiveLayer(layerId as string);
  };

  return (
    <Select
      className={classes.select}
      classes={{ root: classes.selectItem }}
      value={activeLayer}
      onChange={e => handleChange(e)}
    >
      {menuGroup.map(menu => (
        <MenuItem key={menu.id} value={menu.id}>
          {menu.label}
        </MenuItem>
      ))}
    </Select>
  );
}

const styles = () =>
  createStyles({
    select: {
      '&::before': {
        border: 'none',
      },
    },
    selectItem: {
      whiteSpace: 'normal',
      fontSize: 13,
      fontWeight: 300,
      padding: 0,
      marginLeft: 5,
    },
  });

export interface MenuGroupProps extends WithStyles<typeof styles> {
  menuGroup: LayerMenuGroupItem[];
  toggleLayerValue: (layer: LayerType, checked: boolean) => void;
}

export default withStyles(styles)(MenuGroup);
