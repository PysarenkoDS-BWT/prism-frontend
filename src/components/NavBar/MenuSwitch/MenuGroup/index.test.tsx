import React from 'react';
import { render } from '@testing-library/react';
import { LayerMenuGroupItem } from '../../../../config/types';

import MenuGroup from '.';

const props = {
  menuGroup: [{ id: 'id', label: 'label' }] as LayerMenuGroupItem[],
  toggleLayerValue: () => {},
};

test('renders as expected', () => {
  const { container } = render(<MenuGroup {...props} />);
  expect(container).toMatchSnapshot();
});
