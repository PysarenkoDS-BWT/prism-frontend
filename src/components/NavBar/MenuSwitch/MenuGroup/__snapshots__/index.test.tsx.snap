// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders as expected 1`] = `
<div>
  <div
    class="MuiInputBase-root MuiInput-root MuiInput-underline MenuGroup-select-1"
  >
    <div
      aria-haspopup="listbox"
      class="MuiSelect-root MenuGroup-selectItem-2 MuiSelect-select MuiSelect-selectMenu MuiInputBase-input MuiInput-input"
      role="button"
      tabindex="0"
    >
      label
    </div>
    <input
      aria-hidden="true"
      class="MuiSelect-nativeInput"
      tabindex="-1"
      value="id"
    />
    <svg
      aria-hidden="true"
      class="MuiSvgIcon-root MuiSelect-icon"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M7 10l5 5 5-5z"
      />
    </svg>
    <mock-menu
      anchorel="[object HTMLDivElement]"
      id="menu-"
      menulistprops="[object Object]"
      open="false"
      paperprops="[object Object]"
    >
      <mock-menuitem
        aria-selected="true"
        data-value="id"
        role="option"
        selected="true"
      >
        label
      </mock-menuitem>
    </mock-menu>
  </div>
</div>
`;
